### PR TITLE
Improved logging for mail sender new: test-module

### PR DESCRIPTION
**Pull request by glicht**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17165

## Description
Include protocol level debugging when doing test-module. Allows better understanding any issues encountered during configuration.

Includes a change in CommonServer to support using the LOG object as a file like object so it can be used to replace stderr in smtplib.

## Required version of Demisto
All supported

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests - See unit tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Additional changes
Updated common_server_docs.py to include demistomock (as demisto) when generating docs.

### CODE REVIEW

File: Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -558,30 +559,69 @@ class IntegrationLogger(object):
 
     def __init__(self, ):
         self.messages = []  # type: list
-
-    def __call__(self, message):
+        self.write_buf = []  # type: list
+        self.replace_strs = []  # type: list
+        if (not os.getenv('COMMON_SERVER_NO_AUTO_REPLACE_STRS') and hasattr(demisto, 'getParam')
```
Comment (yaakovi) [inline Scripts/CommonServerPython/CommonServerPython.py]: need to add a comment here about `COMMON_SERVER_NO_AUTO_REPLACE_STRS`

File: Integrations/MailSenderNew/MailSenderNew.py [hunk 2]
```diff
@@ -301,28 +305,33 @@ def main():
     global SERVER
     FROM = demisto.getParam('from')
     FQDN = demisto.params().get('fqdn')
+    stderr_org = smtplib.stderr  # type: ignore
     try:
-        SERVER = SMTP(demisto.getParam('host'), int(demisto.getParam('port')), local_hostname=FQDN)
+        if demisto.command() == 'test-module':
```
Comment (yaakovi) [inline Integrations/MailSenderNew/MailSenderNew.py]: why are we triggering this only for test-module?

File: Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -558,30 +559,69 @@ class IntegrationLogger(object):
...
-                self.messages.append("Failed encoding message with error: {}".format(ex))
+                res = "Failed encoding message with error: {}".format(ex)
+        for s in self.replace_strs:
+            res = res.replace(s, '<XX_REPLACED>')
+        return res
+
+    def __call__(self, message):
+        self.messages.append(self.encode(message))
+
+    def add_repalce_strs(self, *args):
```
Comment (yaakovi) [inline Scripts/CommonServerPython/CommonServerPython.py]: logger docstring (for the script helper) should recommend user to call `add_repalce_strs` with the credentials.

we also need to add it to code standard (next to `handle_proxy`)

**Comment by yaakovi** [comment]
@kirbles19 please review CHANGELOG files.

**Comment by kirbles19** [comment]
@yaakovi Done. When approved, please add the `docs-approved` label

File: Integrations/MailSenderNew/MailSenderNew.py (hunk 2 above)
Comment (glicht) [reply Integrations/MailSenderNew/MailSenderNew.py]: for send-mail can affect memory

File: Scripts/CommonServerPython/CommonServerPython.py (hunk 3 above)
Comment (glicht) [reply Scripts/CommonServerPython/CommonServerPython.py]: added comments

File: Scripts/CommonServerPython/CommonServerPython.py (hunk 1 above)
Comment (glicht) [reply Scripts/CommonServerPython/CommonServerPython.py]: added a comment

**Comment by glicht** [comment]
@yaakovi updated the comments

**Pull request merged**